### PR TITLE
Making getters  const

### PR DESCRIPTION
**Pull request by rettinghaus**

### CODE REVIEW

File: src/meidocument.cpp [hunk 1]
```diff
@@ -100,17 +100,17 @@ MeiElement* mei::MeiDocument::getElementById(string id) {
     }
 }
 
-vector<MeiElement*> mei::MeiDocument::getElementsByName(string name) {
+vector<MeiElement*> mei::MeiDocument::getElementsByName(string name) const {
```
Comment (brdvd) [inline src/meidocument.cpp]: Here and in other methods: Passing string arguments by const reference should be preferred, i.e. use `const std::string &name`.

File: src/meielement.cpp [hunk 2]
```diff
@@ -98,23 +98,23 @@ void mei::MeiElement::generateAndSetId() {
     this->setId(out);
 }
 
-const string mei::MeiElement::getId() {
+const string mei::MeiElement::getId() const {
```
Comment (brdvd) [inline src/meielement.cpp:101]: Here and in other methods: Returning a string by value as const doesn't make a lot of sense. Simply returning `string` should be preferred.

File: src/modules/neumesmixins.h [hunk 3]
```diff
@@ -43,7 +43,7 @@ class IneumeLogMixIn {
         void removeForm();
         /** \brief Records the name of the neume.
          */
-        MeiAttribute* getName();
+        MeiAttribute* getName() const;
```
Comment (brdvd) [inline src/modules/neumesmixins.h]: Probably `hasName` should be made `const` as well.

**Review by brdvd — APPROVED** [review]

File: src/meielement.cpp (hunk 2 above)
Comment (rettinghaus) [reply src/meielement.cpp:101]: Perhaps it was originally meant to return by reference? Anyway, I'd leave this for now.

**Comment by rettinghaus** [comment]
@DavidBauer1984 please have another look and merge when appropriate.

**Pull request merged**